### DISCUSSION TITLE
xDrip cant recognise Libre2 sensor & sets to \Libre 1 as "failsafe". No Readings

### DISCUSSION THREAD

**Discussion by nigelkeirross** [NightscoutFoundation/xDrip#3251] [Q&A]
I have been using Libre2 "direct" for about a year. I have just inserted a new Libre2 sensor & starting it as usual method for direct connection. Initially started with Librelink software on another phone, which was turned off. Started OK. After an hour, I stopped the current / old sensor, and did an NFC Scan of new sensor.
The screen message said "scanning" then something that looked like "failsafe" (I didn't quite catch it. I looked in the logs & it says:
Sensor type unknown, returning Libre1 as failsafe 
then
Libre sensorr of type 1 detected.
...
tag read timeout

Has anyone seen this before?

**Comment by Navid200** [comment]
@philTU

**Comment by PhilTU** [comment]
Yes, this arises when the sensor is not correctly identified as a Libre 2 and defaults back to handling it as a Libre 1.

The solution is an update to the OOP2 package to the version from the end of May 2023 and check that xDrip+ is also from that date or later. The "C5" sensor type requires this to be handled correctly, it is appearing progressively in EU markets.

See https://xdrip.readthedocs.io/en/latest/use/OOP/ for download link (Google Drive, you need to be logged in to Google Account) https://drive.google.com/file/d/1f1VHW2I8w7Xe3kSQqdaY3kihPLs47ILS/view

**Comment by nigelkeirross** [comment]
OK many thanks will try this. How do I recognise a "C5" sensor?

**Comment by PhilTU** [comment]
> OK many thanks will try this. How do I recognise a "C5" sensor?

when it doesn't work ;-)  If you look in the logs and perhaps have turned on some extra logging you see PatchInfo or similar from the sensor when it is NFC scanned, and the first non zero data is C5. 

There's discussion on here from when they first appeared with a bit more info. https://github.com/NightscoutFoundation/xDrip/discussions/2845#discussioncomment-5995246

**Reply by nigelkeirross** [reply]
Thanks. I have latest oop2, and more recent xDrip from 7 may 23. I have it running with a bubble bridge, but can't get it to run direct. Will try more tomorrow UK time. It NFC scans ok now, but won't connect and get readings...

**Reply by PhilTU** [reply]
> recent xDrip from 7 may 23

Too old, as I said it needs to be *end of* May to match OOP2 or later

**Reply by nigelkeirross** [reply]
Sorry my mistake, it was 27 May 23. All is working now with the new sensor. I really appreciate the help. Lets hope there is no C6!

**Comment by PhilTU** [comment]
> Lets hope there is no C6!

I think the code was changed to accept more possibilities so hopefully that's now catered for. Glad it's working.
